src/main/java/org/folio/rest/impl/TenantPermsAPI.java

```diff
@@ -250,18 +715,26 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
             Permission foundPerm = null;
             if (!returnList.isEmpty()) {
               foundPerm = returnList.get(0);
+              if (Defined.USER.equals(foundPerm.getDefined())) {
```

Comment:
Would checking `foundPerm.getModuleName()` work?

Reply:
This is the sticky point... if we do that there's no way to determine if it's 
A) a system defined permission lacking the defined context; e.g. during bootstrap, 
or 
B) a user-defined permission.

Reply:
> This is the sticky point... if we do that there's no way to determine if it's
> A) a system defined permission lacking the defined context; e.g. during bootstrap,
> or
> B) a user-defined permission.

Bootstrap is a special case and it has to make some assumptions. For example, if tenant has two modules (A and B) enabled, when bootstrapping the perms for the first module (mod-A), you would not know if mod-A perm provided by Okapi is really a perf defined by mod-A, mod-B or even user by looking at database. But once the bootstrap is done for all modules, we can safely to use moduleName to know whether a permission is system or user defined.

Reply:
Right - above where we identify that the permission provided matches one in the database, except for the moduleName I update the permission right then and there, so that later on (here) we it's no longer ambiguous.